File: backend/src/main/java/com/delivery/delivery/entity/Delivery.java
package com.delivery.delivery.entity;

import com.delivery.store.entity.Store;
import com.delivery.deliveryslot.entity.DeliverySlot;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "deliveries")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Delivery {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private DeliveryStatus status;

    @Column(nullable = false)
    private String deliveryCompany;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "store_id", nullable = false)
    private Store store;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "delivery_slot_id", nullable = false)
    private DeliverySlot deliverySlot;
}

File: frontend/src/interfaces/Delivery.ts
import {DeliverySlot} from "./DeliverySlot"
import {Store} from "./Store"

export interface Delivery {
    id: number;
    address: string;
    status: DeliveryStatus;
    deliveryCompany: string;
    store: Store
    deliverySlot?: DeliverySlot
}

export enum DeliveryStatus {
    PENDING = 'PENDING',
    IN_PROGRESS = 'IN_PROGRESS',
    DELIVERED = 'DELIVERED',
    CANCELLED = 'CANCELLED',
}